### DISCUSSION TITLE
The images in xls format are turning into a blank cell. (Birt Runtime 4.15)

### DISCUSSION THREAD

**Discussion by LeonardoIPM** [eclipse-birt/birt#1718] [Q&A] [answered]
When generating a pdf or docx report by ReportEngine, our images are ok. But when generating a xls file, the image just gone and the cell gets a diagonal border inside it.
I tried to configure the RenderOption and EXCELRenderOption, but unsuccessfully.
With some search, it seems that I need to use another emitter, but inside birtruntime has the spudsoft one, and it seems to be the best one. So... I think it's just a wrong configuration, what else should I do?

My configuration:

```
RenderOption optGeral = new RenderOption(); //tried EXCELRenderOption also
//optGeral.setEmitterID("uk.co.spudsoft.birt.emitters.excel.XlsEmitter"); breaks xls generations, but not xlsx
optGeral.setOutputFormat(format); // (format can be xls and xlsx)
optGeral.setOption(ExcelEmitter.DEBUG, true);
optGeral.setOption(ExcelEmitter.SINGLE_SHEET, true);
optGeral.setOption(ExcelEmitter.SINGLE_SHEET_PAGE_BREAKS, true);
```

I tried to set a ImageHandler, unsuccessfully too...

**Comment by speckyspooky** [comment]
Have you a screen of it that we can see the impact or a small report example?
I fixed a image issue for spudsoft-excel at the latest version for BIRT 4.16. May be you can test your report with it.

**Comment by LeonardoIPM** [comment]
Thanks for the reply!

Unfortunately, I can't provide a report or a picture of the issue with 4.15 BIRT, just an old image from 4.4.2, but the result is almost the same:

![report](https://github.com/eclipse-birt/birt/assets/167772501/921bbff6-f3ae-4b35-8e88-c6b1060a45d8)

In pdf my image is fine, but in xls the cell is blank.
I draw a black line simulating the diagonal border in xls cell in the 4.15 version.

Edit: But if it helps, the xml has the following image structure:

```
<cell>
      <property name="textAlign">left></property>
      <image>
            <property name="source">embed</property>
            <property name="imageName">brasao</property>
       </image>
</cell>

...

<list-property name="images"><structure>
<property name="name">brasao</property>
<property name="data">(the image data)></property>
<property name="type">image/jpeg</property>
</structure></list-property></report>
```

**Reply by speckyspooky** [reply]
I created based on your example screen a static test version of the excel.
My result looks good - excel with image. I added for you my test report. Please try it and post the result.

It could by a topic with the image because the image handling is implemented for each output emitter separately.

[birt_415_image_head.zip](https://github.com/user-attachments/files/15749372/birt_415_image_head.zip)

![grafik](https://github.com/eclipse-birt/birt/assets/41593722/8a578162-f043-4874-9c4d-22f694e7c630)

**Reply by speckyspooky** [reply]
May be you would be able to share the image.
Sometimes images causes trouble internal at runtime when the emitter tries to handle these.
So I would be able to test the image from my side to.

**Comment by LeonardoIPM** [comment]
So, I tried your report and everything is fine.
But, if I switch the uri image for an embedded image, then it fails to render the image again in my application. (Previewing it on Birt Client All in One, the image has no problems).

**Comment by speckyspooky** [comment]
@LeonardoIPM 
Could you please provide your used image to verify it. In best case I can get the error on my side too.

The problem has nothing to do by default with embedded images - attached "ipm"-logo-based embedded image which looks ok.
[embedded-image.zip](https://github.com/user-attachments/files/15777484/embedded-image.zip)

**Comment by LeonardoIPM** [comment]
Here is our image, but even your images aren't rendered here. Your last report worked in Birt Client, but when I render it using our application, the image just turn a "\" border in the cell. Other formats are all ok.
And you are right, that's nothing about embedded images, the smile face image wasn't renderized also.
[download.zip](https://github.com/user-attachments/files/15777993/download.zip)

**Reply by speckyspooky** [reply]
I used your image and based on the all-in-one-designer I got the correct image into excel.

According to your and my test results, it seems that there is a dependency/configuration into your application which cause this topic.

**Reply by LeonardoIPM** [reply]
Do you know which dependencies, or how should I configure my application?
For xls/xlsx, I am using RenderOptions with just the configurations of the first post.
Did you try with org.eclipse.birt.runtime_4.15.0?

**Reply by speckyspooky** [reply]
Of course, I have used BIRT 4.14, 4.15, 4.16 to test the images.
The question is in which way you have implemented the runtime?
Do you use the runtime-package of birt or the runtime-OSGi-packages of BIRT?

**Reply by LeonardoIPM** [reply]
Thanks for the patience!
We downloaded the birt-runtime-4.15.0-202403270652.zip, took the ReportEngine libs and put them in our project. We already had the 4.4 version dependencies, then manually we removed the ones that seems cause conflict. (Both 4.4 and 4.15 version don't render images in xls).

**Reply by LeonardoIPM** [reply, marked as answer]
It was my bad actually.
I was using XLS emitter and not SPUDSOFT_XLS emitter because I hadn't set the option STRUCTURED_HEADER and my header just wasn't rendered. But the XLS one does not support images. 
Now the problem is the SPUDSOFT formatting, Maybe I need to set new configurations, IDK. But I will open a new discussion about it.
